src/main/java/com/netflix/imflibrary/utils/DOMNodeObjectModel.java

```diff
@@ -724,12 +743,31 @@ public Set<String> getFieldsAsStringRecursive(String name) {
         return values;
     }
 
+    /**
+     * A method to obtain set of Integer values for a field within DOMNodeObjectModel
+     * @param name the LocalName for the field
+     * @return Returns a list of field values
+     */
+    public List<Integer> getFieldsAsInteger(String name) {
```

Comment:
Should `getFieldsAsInteger` internally call `getFieldsAsStringRecursive`, the same way the methods for [UUID](https://github.com/Netflix/photon/blob/beda2484b5df63b0383a463841eaa8da6c5d14f7/src/main/java/com/netflix/imflibrary/utils/DOMNodeObjectModel.java#L774) and [UL](https://github.com/Netflix/photon/blob/beda2484b5df63b0383a463841eaa8da6c5d14f7/src/main/java/com/netflix/imflibrary/utils/DOMNodeObjectModel.java#L753) fields are implemented?

Reply:
I was not sure exactly what `getFieldsAsStringRecursive()` was intended for (what does "recursive" refer to?_, so did not use it, but happy to do so if it you recommend doing so.

Reply:
You are right. The "recursive" part is a bit strange... As far as I can tell, there aren't any cases where any of the properties that use this are ever _nested_ in a way that would require recursion. Certainly not for your use-cases.
I'll see if I can dig up any additional context about the original design for making this "recursive". As far as I can tell, it may have just been a lazy solution, so you can start at the top of a tree, and find a certain field by name. This eliminates the need to have to manually locate the parent node I guess.

I think its okay if you just keep using the code the way you _currently_ have it.